pkg/spdk/util.go

```diff
@@ -14,13 +15,43 @@ import (
 	"github.com/longhorn/go-spdk-helper/pkg/nvme"
 
 	commonns "github.com/longhorn/go-common-libs/ns"
+	commontypes "github.com/longhorn/go-common-libs/types"
 	commonutils "github.com/longhorn/go-common-libs/utils"
 	spdkclient "github.com/longhorn/go-spdk-helper/pkg/spdk/client"
 	spdktypes "github.com/longhorn/go-spdk-helper/pkg/spdk/types"
 	helpertypes "github.com/longhorn/go-spdk-helper/pkg/types"
 	helperutil "github.com/longhorn/go-spdk-helper/pkg/util"
 )
 
+func connectNVMeTarget(srcIP string, srcPort int32, maxRetries int, retryInterval time.Duration) (string, string, error) {
+	executor, err := helperutil.NewExecutor(commontypes.ProcDirectory)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "failed to create executor")
+	}
+
+	subsystemNQN := ""
+	controllerName := ""
+	for r := 0; r < maxRetries; r++ {
```

Comment:
Should we exit the `for loop` earlier when both `DiscoverTarget` and `ConnectTarget` succeed?
If yes, can you update `exposeSnapshotLvolBdev()` as well?

Reply:
fixed.